lib/rubocop/cop/lint/redundant_type_conversion.rb

```diff
@@ -211,7 +223,14 @@ def constructor?(node, receiver)
           matcher = CONSTRUCTOR_MAPPING[node.method_name]
           return false unless matcher
 
-          public_send(matcher, receiver)
+          public_send(matcher, receiver) && !constructor_suppresses_exceptions?(receiver)
+        end
+
+        def constructor_suppresses_exceptions?(receiver)
+          # If the constructor suppresses exceptions with `exception: false`, it is possible
+          # it could return `nil`, and therefore a chained conversion is not redundant.
+
+          receiver.arguments.any? { |arg| exception_false?(arg) }
```

Comment:
The blank line appears redundant.

```suggestion
          # If the constructor suppresses exceptions with `exception: false`, it is possible
          # it could return `nil`, and therefore a chained conversion is not redundant.
          receiver.arguments.any? { |arg| exception_false?(arg) }
```